### DISCUSSION TITLE
Requirements for 2.0 release

### DISCUSSION THREAD

**Discussion by dhrogers** [cinemascience/pycinema#26] [General]
**Requirements:**
- [x] Rename 'designer' script to 'cinema'
- [x] Rename 'Designer' class to 'Theater'
- [x] Add colors, etc. for 'light' UI implementation (currently only 'dark' is supported well)
- Functional requirements
    - [x] 'view' and 'explorer' workspaces must work with all datasets in repository
    - [x] default image view should be a grid
    - [x] build/save/reload must work with all datasets in repository
    - [x] ambient occlusion node will be added to graph for default workspaces, with no UI controls, but with defaults
    - [x] 'cinema view/explorer' will pop up a dialogue box so user can select database
    - [ ] 'view' must handle up to four input databases. Assumption is that all parameters have the same values
- [ ] In general, anything that takes a table as input and has a table as output should have an 'ignore' parameter. The meaning of the parameter is 'columns from the input table that I do not operate on, but that are passed on to the output table'

**Table View**
- Add 'ignore' input parameter, with behavior to mimic that of ParallelCoordinatesView
- Add 'id' value to 'ignore' input parameter as a default

**Questions**
- 'Explorer' workspace: should TableView columns match ParallelCoordinatesView columns?
 
**New Features**
- [v2.0] Handle Cinema databases that do not have FILE or other assets (are pure parameters)
- [vx.x] Open and view csv file with new 'Table' workspace. This would open a csv file and create a parallel coordinates view and table view with all columns shown.
- [vx.x] 'python' node that runs arbitrary python code in inputs, and can pass data to output
- [vx.x] Parallel Coordinates view: Make individual lines pickable (new feature)

**Comment by dbanesh** [comment]
**Questions**

- How do I unlink filters that have been linked in node view? 
- Does opening anything using 'Open' automatically default to a 'Cinema View' configuration?

**New Features**

- An 'x' at the top right corner to delete filters from the node view (the current method is not obvious)
- The H and V for splits is a bit confusing. Can we use the similar icons as Paraview to make it super clear which way the split will be?

**Issues/Possible Bugs**

- When loading Cinema databases from 'Open', selecting any folder, regardless of whether it's a cdb or not, seems to work. This might be ok if not for the fact that double clicking when navigating to my required folder automatically opens up the wrong folder. 
- Deleting the color mapping view seems to delete the data from the Image View as well. Is this intended or a bug? There seems to be no way to get it back once deleted. Reopenning the color mapping view lists the channel as nothing
- Adding a parallel coordinates view after opening a data in the 'Cinema View' configuration does not seem to show anything. It may not be linked as we expect and we should look into this further. 
- "cinema path/to/db" does not seem to work (opens the cinema but does not seem to have a db loader)
- "cinema explorer path/to/sphere.cdb" produces some errors. This may be a light/dark format issue. Should be explored further.

**Comment by dhrogers** [comment]
- Should there be a table window in the 'view' and 'explorer' workspaces? This is an addition to the 'normal' applications.

**Comment by dhrogers** [comment]
- created readthedocs repository: pycinema.readthedocs.io
